agent-lib/src/test/java/org/terracotta/angela/client/DummyKitResolver.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * The contents of this file are subject to the Terracotta Public License Version
+ * 2.0 (the "License"); You may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://terracotta.org/legal/terracotta-public-license.
+ *
+ * Software distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for
+ * the specific language governing rights and limitations under the License.
+ *
+ * The Covered Software is Angela.
+ *
+ * The Initial Developer of the Covered Software is
+ * Terracotta, Inc., a Software AG company
+ */
+
+package org.terracotta.angela.client;
+
+import org.terracotta.angela.KitResolver;
+import org.terracotta.angela.common.tcconfig.License;
+import org.terracotta.angela.common.topology.LicenseType;
+import org.terracotta.angela.common.topology.PackageType;
+import org.terracotta.angela.common.topology.Version;
+
+import java.net.URL;
+import java.nio.file.Path;
+
+/**
+ * @author Aurelien Broszniowski
+ */
+
+public class DummyKitResolver extends KitResolver {
```

Comment:
Instead of creating single-use kit resolvers, we could add a new method `getServiceLoader` in  `LocalKitManager` which return a `ServiceLoader` instance. This method can be stubbed in the test code to return a mocked instance of `ServiceLoader` with whatever behavior we want from it.

Reply:
I found an easier way, I pass a null Distribution and the serviceloader mechanism will not be called